test/fork/watch-restart.test.js

```diff
@@ -86,6 +86,41 @@ describe('nodemon fork child restart', function () {
     });
   });
 
+  it('should happen only once if delay option is set', function (done) {
+    var restartCount = 0;
+    fs.writeFileSync(tmpjs, 'true;');
+
+    var p = run('--ext js --delay 1' + appjs, {
+      error: function (data) {
+        p.send('quit');
+        cleanup(p, done, new Error(data));
+      }
+    });
+
+    p.on('message', function (event) {
+      if (event.type === 'start') {
+        setTimeout(function () {
```

Comment:
These two timeouts get set twice - but once the script restarts, it triggers the start event again, which causes this to reset. There's also often issues in Travis when touching files and nodemon picking up the change takes about 1 second, so these happening so closely to each other were hiding issues (along with the 1.5s timeout to check the status).